packages/clerk-js/src/ui/components/SignIn/SignInStart.tsx

```diff
@@ -73,33 +83,40 @@ export function _SignInStart(): JSX.Element {
     setIdentifierAttribute(
       i => identifierAttributes[(identifierAttributes.indexOf(i) + 1) % identifierAttributes.length],
     );
-    identifierField.setValue('');
   };
 
-  const switchToPhoneInput = (value?: string) => {
+  const switchToPhoneInput = () => {
     setIdentifierAttribute('phone_number');
-    identifierField.setValue(value || '');
   };
 
   // switch to the phone input (if available) if a "+" is entered
   // (either by the browser or the user)
   // this does not work in chrome as it does not fire the change event and the value is
   // not available via js
-  React.useLayoutEffect(() => {
+  useLayoutEffect(() => {
     if (
       identifierField.value.startsWith('+') &&
       identifierAttributes.includes('phone_number') &&
       identifierAttribute !== 'phone_number' &&
       !hasSwitchedByAutofill
     ) {
-      switchToPhoneInput(identifierField.value);
+      switchToPhoneInput();
       // do not switch automatically on subsequent autofills
       // by the browser to avoid a switch loop
       setHasSwitchedByAutofill(true);
     }
   }, [identifierField.value, identifierAttributes]);
 
-  React.useEffect(() => {
+  useLayoutEffect(() => {
+    if (identifierAttribute === 'phone_number' && identifierField.value.startsWith('+')) {
```

Comment:
@desiprisg What are we trying to achieve with this? Is it about the country code picker?